spectral_cube/masks.py

```diff
@@ -314,13 +323,52 @@ class CompositeMask(MaskBase):
     """
 
     def __init__(self, mask1, mask2, operation='and'):
+        if isinstance(mask1, np.ndarray) and isinstance(mask2, MaskBase) and hasattr(mask2, 'shape'):
+            if not is_broadcastable_and_smaller(mask1.shape, mask2.shape):
+                raise ValueError("Mask1 shape is not broadcastable to Mask2 shape: "
+                                 "%s vs %s" % (mask1.shape, mask2.shape))
+            mask1 = BooleanArrayMask(mask1, mask2._wcs, shape=mask2.shape)
+        elif isinstance(mask2, np.ndarray) and isinstance(mask1, MaskBase) and hasattr(mask1, 'shape'):
+            if not is_broadcastable_and_smaller(mask2.shape, mask1.shape):
+                raise ValueError("Mask2 shape is not broadcastable to Mask1 shape: "
+                                 "%s vs %s" % (mask2.shape, mask1.shape))
+            mask2 = BooleanArrayMask(mask2, mask1._wcs, shape=mask1.shape)
+
+        # both entries must have compatible, which effectively means
+        # equal, WCSes.  Unless one is a function.
+        if hasattr(mask1, '_wcs') and hasattr(mask2, '_wcs'):
+            mask1._validate_wcs(new_data=None, wcs=mask2._wcs)
+            # In order to composite composites, they must have a _wcs defined.
+            # (maybe this should be a property?)
+            self._wcs = mask1._wcs
+        elif hasattr(mask1, '_wcs'):
+            # if one mask doesn't have a WCS, but the other does, the
+            # compositemask should have the same WCS as the one that does
+            self._wcs = mask1._wcs
+        elif hasattr(mask2, '_wcs'):
+            self._wcs = mask2._wcs
+
```

Comment:
Can the `else` case ever happen? Presumably not if someone is masking via a `SpectralCube`. If for some reason you used `CompositeMask` on its own and gave it two numpy arrays, it wouldn't fail until a `_validate_wcs` call is triggered. We could raise a WCS error just in case.

Reply:
If you do something like `cube.with_mask(FunctionMask(lambda x: x>5) & (cube > 5))` (which is redundant...), the first mask would not have a `_wcs` attribute, but the second would

Reply:
I'm referring to the case where neither has a WCS object. It would only happen (I think) if you're creating a `CompositeMask` directly from two numpy arrays. I can't think of _why_ you would ever want to do this, so adding an error in the else case is largely redundant. You're welcome to ignore this point.

Reply:
my bad, you're right.  You could get it from composing two `FunctionMask`s though

Reply:
The fail case here is just to leave `_wcs` undefined, though. I think that's actually the desired behavior since I'm doing `hasattr` tests

Reply:
Sounds good!